### PR TITLE
Fixes #1194 & #1196: AdminControls - Tablet (Portrait) (Landscape) (Lowfi)  

### LINKED ISSUE #1194: AdminControls - Tablet (Landscape) (Lowfi)
Mocks: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/ed5480b9-cee5-4b28-845f-f0bf5d965653/  

Implement low-fi UI for **AdminControls** tablet landscape mode

**Target PR date**: 26 July 2020
**Target completion date**: 30 July 2020

### LINKED ISSUE #1196: AdminControls - Tablet (Portrait) (Lowfi)
Mocks: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/f55f5e39-3889-4900-a22a-4f6e531a51ca/  

Implement low-fi UI for **AdminControls** tablet portrait mode

**Target PR date**: 26 July 2020
**Target completion date**: 30 July 2020

### PR DESCRIPTION
**Pull request by MohamedMedhat1998**
## Explanation

- Fixes #1194 
- Fixes #1196 
- Implemented low-fi UI for **AdministiratorControls** for tablet for both landscape and portrait
- Mocks: [land](https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/ed5480b9-cee5-4b28-845f-f0bf5d965653/) - [port](https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/f55f5e39-3889-4900-a22a-4f6e531a51ca/)
## Screenshots
screenshot-٢٠٢٠-٠٨-٢٤_٢٠ ١٥ ٢١ ٤٨٥
screenshot-٢٠٢٠-٠٨-٢٤_٢٠ ١٥ ١٨ ٧٢٩
screenshot-٢٠٢٠-٠٨-٢٤_٢٠ ١٤ ٥٧ ٠٧٣
screenshot-٢٠٢٠-٠٨-٢٤_٢٠ ١٤ ٢١ ٢٢٧
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@MohamedMedhat1998 For bazel CI fail, check this: https://github.com/oppia/oppia-android/pull/1705#issuecomment-679284547 and my solution in that PR corresponding to that.

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivity.kt [hunk 1]
```diff
@@ -50,4 +66,27 @@ class AdministratorControlsActivity :
...
+  override fun loadAppVersion() {
+    lastLoadedFragment = APP_VERSION_FRAGMENT
+    administratorControlsActivityPresenter
+      .setExtraControlsTitle(getString(R.string.administrator_controls_app_version))
+    administratorControlsActivityPresenter.loadAppVersion()
+  }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    val titleTextView = findViewById<TextView>(R.id.extra_controls_title)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivity.kt]: This logic should be shifted to presenter. Try to keep `Activity` keep as much as possible.

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt [hunk 2]
```diff
@@ -15,15 +19,34 @@ class AdministratorControlsActivityPresenter @Inject constructor(
...
+  fun handleOnCreate(extraControlsTitle: String?, lastLoadedFragment: String) {
     activity.setContentView(R.layout.administrator_controls_activity)
     setUpNavigationDrawer()
-    if (getAdministratorControlsFragment() == null) {
-      activity.supportFragmentManager.beginTransaction().add(
-        R.id.administrator_controls_fragment_placeholder,
-        AdministratorControlsFragment()
-      ).commitNow()
+    val titleTextView =
+      activity.findViewById<TextView>(R.id.extra_controls_title)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt]: I think you should control this into a data binding by introducing `<layout>` in xml file.

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt [hunk 3]
```diff
@@ -53,4 +76,22 @@ class AdministratorControlsActivityPresenter @Inject constructor(
...
+
+  fun loadAppVersion() {
+    activity.supportFragmentManager.beginTransaction().add(
+      R.id.administrator_controls_fragment_multipane_placeholder,
+      AppVersionFragment()
+    ).commitNow()
+  }
+
+  fun setExtraControlsTitle(title: String) {
+    activity.findViewById<TextView>(R.id.extra_controls_title).text = title
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt]: Change this using data binding / viewmodel

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsFragment.kt [hunk 4]
```diff
@@ -8,11 +8,23 @@ import android.view.ViewGroup
 import org.oppia.app.fragment.InjectableFragment
 import javax.inject.Inject
 
+private const val IS_MULTIPANE_KEY = "IS_MULTIPANE_KEY"
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsFragment.kt]: Shift this to the companion object.

File: app/src/main/java/org/oppia/app/administratorcontrols/LoadAppVersionListener.kt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package org.oppia.app.administratorcontrols
+
+interface LoadAppVersionListener {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/administratorcontrols/LoadAppVersionListener.kt:4]: Add KDoc

File: app/src/main/java/org/oppia/app/administratorcontrols/LoadProfileListListener.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package org.oppia.app.administratorcontrols
+
+interface LoadProfileListListener {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/administratorcontrols/LoadProfileListListener.kt:4]: Add KDoc

**Review by rt4914 — CHANGES_REQUESTED** [review]
Approach looks good. I have suggested some changes.

Thanks.

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivity.kt (hunk 1 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivity.kt]: Done

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt (hunk 2 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt]: Done

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt (hunk 3 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityPresenter.kt]: Done

File: app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsFragment.kt (hunk 4 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsFragment.kt]: Done

File: app/src/main/java/org/oppia/app/administratorcontrols/LoadAppVersionListener.kt (hunk 5 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/administratorcontrols/LoadAppVersionListener.kt:4]: Done

File: app/src/main/java/org/oppia/app/administratorcontrols/LoadProfileListListener.kt (hunk 6 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/administratorcontrols/LoadProfileListListener.kt:4]: Done

**Comment by MohamedMedhat1998** [comment]
> @MohamedMedhat1998 For bazel CI fail, check this: [#1705 (comment)](https://github.com/oppia/oppia-android/pull/1705#issuecomment-679284547) and my solution in that PR corresponding to that.

It is already added in the develop. I just updated my branch with the latest develop

**Comment by MohamedMedhat1998** [comment]
> > @MohamedMedhat1998 For bazel CI fail, check this: [#1705 (comment)](https://github.com/oppia/oppia-android/pull/1705#issuecomment-679284547) and my solution in that PR corresponding to that.
> 
> It is already added in the develop. I just updated my branch with the latest develop

Oh, I think I understood what I should do.

File: app/BUILD.bazel [hunk 7]
```diff
@@ -43,6 +43,8 @@ Place your file here if:
 LISTENERS = [
     "src/main/java/org/oppia/app/administratorcontrols/RouteToAppVersionListener.kt",
     "src/main/java/org/oppia/app/administratorcontrols/RouteToProfileListListener.kt",
+    "src/main/java/org/oppia/app/administratorcontrols/LoadAppVersionListener.kt",
+    "src/main/java/org/oppia/app/administratorcontrols/LoadProfileListListener.kt",
```
Comment (rt4914) [inline app/BUILD.bazel:45]: Please keep them in alphabetical order.

**Review by rt4914 — APPROVED** [review]
One minor change. Also, have a look at build error. Thanks.

File: app/BUILD.bazel (hunk 7 above)
Comment (rt4914) [reply app/BUILD.bazel:45]: If build passes, you can resolve this comment on your own.

Comment (MohamedMedhat1998) [reply app/BUILD.bazel:45]: It is still failing, I don't know why.  
I have checked the details of the error but I can't figure out what the problem is.  
```
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
##[error]Process completed with exit code 1.
```

Comment (MohamedMedhat1998) [reply app/BUILD.bazel:45]: Worked after adding `artifact("androidx.drawerlayout:drawerlayout:1.1.0"),` to BUILD.bazel as you suggested

**Pull request merged**